airflow/sensors/sql_sensor.py

```diff
@@ -98,4 +102,6 @@ def poke(self, context):
                 return self.success(first_cell)
             else:
                 raise AirflowException("self.success is present, but not callable -> {}".format(self.success))
-        return str(first_cell) not in ('0', '')
+        if self.allow_null:
+            return str(first_cell) not in ('0', '')
+        return str(first_cell) not in ('0', '', 'None')
```

Comment:
```suggestion
        return first_cell is not None and str(first_cell) not in ('0', '')
``

Reply:
(And although not  a new bug, this won't work if the type is floating point, that would give us 0.0 as a value)